### DISCUSSION TITLE
URQL results update in dev but not in production build

### DISCUSSION THREAD

**Discussion by jjcroftiv** [urql-graphql/urql#2902] [Q&A] [answered]
I have an app where I'm adding items to a list. When running in dev mode, everything works as expected: I execute the mutation, the query updates, and I see the new item. But if I build the app and run the optimized version, the mutation works, but the view doesn't update. Any thoughts as to what might be going on?

**Comment by JoviDeCroock** [comment, marked as answer]
Are you seeing any console warnings in development, because this could be one of our developer warnings that's triggering which don't trigger in production

**Reply by jjcroftiv** [reply]
In fact, I am. "Invalid undefined: The field at `eventsByAspectEventAspectIdAndEventIdList` is `undefined`, but the GraphQL query expects a selection set for this field." It doesn't cause an issue, so I hadn't noticed it. I'll see if resolving this warning resolves the production behavior.

**Reply by JoviDeCroock** [reply]
That should in fact fix the issue, maybe we should fail there instead of performing a `continue`

**Reply by JoviDeCroock** [reply]
Hey @jjcroftiv did this solve your issue? You could be calling `updateQuery`, ... where you are passing `undefined` for that field

**Reply by jjcroftiv** [reply]
@JoviDeCroock Sorry for the delay, now back at the desk. I resolved the warning, and the issue, but the two weren't related. The actual issue was related to cache management in the mutation handler. In the handler, when I received the created record, I did a data push of the new record. In my local environment, the record was added at the top of the list, as expected, since the query was ordered by date descending. But in the production build, it added it to the end of the list, where I didn't see it. I resolved it by using data unshift to ensure it ended up at the front of the array. Let me know if you want any more details, and thank you for your engagement!
